cpp/tests/test_ide_parameters_io.cpp

```diff
@@ -52,35 +60,50 @@ TEST(TestIDEParametersIo, RKIcompareWithPreviousRun)
+        model.parameters.get<mio::isecir::TransitionProbabilities>()[group] = vec_prob;
+    }
+    mio::ContactMatrixGroup contact_matrix = mio::ContactMatrixGroup(1, num_agegroups);
+    contact_matrix[0] = mio::ContactMatrix(Eigen::MatrixXd::Constant(num_agegroups, num_agegroups, 10.));
     model.parameters.get<mio::isecir::ContactPatterns>() = mio::UncertainContactMatrix(contact_matrix);
 
     mio::ConstantFunction constfunc(1.0);
     mio::StateAgeFunctionWrapper prob(constfunc);
-    model.parameters.set<mio::isecir::TransmissionProbabilityOnContact>(prob);
-    model.parameters.set<mio::isecir::RelativeTransmissionNoSymptoms>(prob);
-    model.parameters.set<mio::isecir::RiskOfInfectionFromSymptomatic>(prob);
+
+    for (auto group = mio::AgeGroup(0); group < mio::AgeGroup(num_agegroups); ++group) {
+        model.parameters.get<mio::isecir::TransmissionProbabilityOnContact>()[group] = prob;
+        model.parameters.get<mio::isecir::RelativeTransmissionNoSymptoms>()[group]   = prob;
+        model.parameters.get<mio::isecir::RiskOfInfectionFromSymptomatic>()[group]   = prob;
+    }
 
     // Calculate initialization.
     auto status =
-        mio::isecir::set_initial_flows(model, dt, mio::path_join(TEST_DATA_DIR, "cases_all_germany.json"), start_date);
+        mio::isecir::set_initial_flows(model, dt, mio::path_join(TEST_DATA_DIR, "cases_all_age_ma7.json"), start_date);
 
     ASSERT_THAT(print_wrap(status), IsSuccess());
 
-    // Compare with previous run.
-    auto compare = load_test_data_csv<ScalarType>("ide-parameters-io-compare.csv");
+    //Compare with previous run.
+    auto compare = load_test_data_csv<ScalarType>("ide-ageres-parameters-io-compare.csv");
 
-    EXPECT_NEAR(model.m_populations.get_value(0)[(Eigen::Index)mio::isecir::InfectionState::Dead], 6.3, 1e-4);
-    EXPECT_NEAR(model.m_total_confirmed_cases, 70, 1e-4);
+    std::vector<ScalarType> deaths = {
+        1, 2.471428571455, 26.34999999999, 603.621428571465, 3972.41428571431, 7668.84999999998};
 
+    std::vector<ScalarType> total_confirmed_cases = {10269.2857142857, 29615.8571428571, 185321.571428571,
+                                                     215386.428571429, 77163.5714285714, 35588.4285714286};
+
+    for (mio::AgeGroup group = mio::AgeGroup(0); group < mio::AgeGroup(num_agegroups); ++group) {
+        int Di = model.get_state_flat_index((Eigen::Index)mio::isecir::InfectionState::Dead, group);
+        EXPECT_NEAR(model.m_populations.get_value(0)[Di], deaths[size_t(group)], 1e-4);
+        EXPECT_NEAR(model.m_total_confirmed_cases[group], total_confirmed_cases[size_t(group)], 1e-4);
+    }
+    mio::isecir::Simulation sim(model, dt);
     ASSERT_EQ(compare.size(), static_cast<size_t>(model.m_transitions.get_num_time_points()));
     for (size_t i = 0; i < compare.size(); i++) {
         ASSERT_EQ(compare[i].size(), static_cast<size_t>(model.m_transitions.get_num_elements()) + 1) << "at row " << i;
```

Comment:
Can you adapt this as in the test_ide_secir_ageres and only do the comparison for the last time point?